Modules/IO/VTK/src/itkVTKImageIO.cxx

```diff
@@ -858,9 +858,7 @@ VTKImageIO::WriteSymmetricTensorBufferAsBinary(std::ostream &                 os
   std::streamsize bytesRemaining = num;
   const SizeType  componentSize = this->GetComponentSize();
   SizeType        pixelSize;
-  char            zero[1024];
-
-  memset(zero, 0, 1024);
+  constexpr char  zero[1024]{};
```

Comment:
@blowekamp Thanks for sharing your considerations regarding the possible performance effect of this commit. Why do you think `T var{}` might be slower than `T var; memset(&var, 0, sizeof(T))`?

In this particular case,  `constexpr char zero[1024]{}` is probably _faster_ than memset, because the `{}` initialization may be performed at compile-time. (We could even _enforce_ compile-time evaluation by making the variable `static constexpr`, but I doubt if that's worth it, in this particular case.)

Reply:
As I recall my previous performance experiments were likely on large image buffers where a significant difference was measured. As I have seen a few of these changes preferring the initialization, I was querying in general how does the performance compare.

Quickly scanning these cases, it does not appear to be many large buffers in performance critical code. I don't think the are are carefully optimized cases.

The using the default initialize is cleaner, and more standardize code. In general, I'm in favor of it.

Reply:
Thank you for the discussion and deep dive.